app/ide-desktop/lib/dashboard/src/providers/InputBindingsProvider.tsx

```diff
@@ -28,19 +28,27 @@ export interface InputBindingsProviderProps extends React.PropsWithChildren<obje
 export default function InputBindingsProvider(props: InputBindingsProviderProps) {
   const { inputBindings: inputBindingsRaw, children } = props
   const [inputBindings, setInputBindings] = React.useState(
-    () => inputBindingsRaw ?? inputBindingsModule.createBindings()
+    () => inputBindingsRaw ?? inputBindingsModule.createBindings(false)
   )
 
   React.useEffect(() => {
-    setInputBindings(inputBindingsRaw ?? inputBindingsModule.createBindings())
+    inputBindings.register()
+  }, [inputBindings])
+
+  React.useEffect(() => {
+    setInputBindings(oldBindings => {
+      oldBindings.unregister()
+      return inputBindingsRaw ?? inputBindingsModule.createBindings(false)
+    })
   }, [inputBindingsRaw])
 
   return (
     <InputBindingsContext.Provider value={inputBindings}>{children}</InputBindingsContext.Provider>
   )
 }
```

Comment:
The cleanup here doesn't look right. I believe that once the `InputBindingsProvider` component is unmounted, the last state of `inputBindings` will not be properly deregistered.

Would this simpler version work?
```ts
  const { inputBindings, children } = props
  
   React.useEffect(() => {
    inputBindings.register()
    return () => inputBindings.unregister()
  }, [inputBindings])
  ```

Reply:
oh... yeah that should work, thanks